.castor.php

```diff
@@ -0,0 +1,441 @@
+        'docker_compose_files' => [
+            'docker-compose.yml',
+            'docker-compose.worker.yml',
+        ],
+        'dinghy' => false,
+        'macos' => false,
+        'power_shell' => false,
+        'user_id' => 1000,
+        'root_dir' => __DIR__,
+        'start_workers' => false,
+        'env' => $_SERVER['CI'] ?? false ? 'ci' : 'dev',
+    ];
+
+    if (file_exists($c['root_dir'] . '/infrastructure/docker/docker-compose.override.yml')) {
+        $c['docker_compose_files'][] = 'docker-compose.override.yml';
+    }
+
+    $composerCacheDir = null;
+
+    try {
+        $composerCacheDir = trim(exec(['composer', 'global', 'config', 'cache-dir', '-q'], quiet: true, pty: false)->getOutput());
+    } catch (ExceptionInterface) {
+    }
+
+    if ($composerCacheDir) {
+        $c['composer_cache_dir'] = trim($composerCacheDir);
+    }
+
+    $platform = strtolower(php_uname('s'));
+
+    if (str_contains($platform, 'darwin')) {
+        $c['macos'] = true;
+    }
+
+    return new Context($c, pty: 'dev' === $c['env']);
+}
+
+function docker_compose_run(
+    Context $c,
+    string $runCommand,
+    string $service = 'builder',
+    string $user = 'app',
+    bool $noDeps = true,
+    string $workDir = null,
+    bool $portMapping = false,
+    bool $withBuilder = true,
+): Process {
+    $command = [
+        'run',
+        '--rm',
+        '-u', $user,
+    ];
+
+    if ($noDeps) {
+        $command[] = '--no-deps';
+    }
+
+    if ($portMapping) {
+        $command[] = '--service-ports';
+    }
+
+    if (null !== $workDir) {
+        $command[] = '-w';
+        $command[] = $workDir;
+    }
+
+    $command[] = $service;
+    $command[] = '/bin/sh';
+    $command[] = '-c';
+    $command[] = "exec {$runCommand}";
+
+    return docker_compose($c, $command, withBuilder: $withBuilder);
+}
+
+function docker_compose(Context $c, array $subCommand, bool $withBuilder = false): Process
+{
+    $domains = [$c['root_domain'], ...$c['extra_domains']];
+    $domains = '`' . implode('`, `', $domains) . '`';
+
+    $c = $c
+        ->withTimeout(null)
+        ->withEnvironment([
+            'PROJECT_NAME' => $c['project_name'],
+            'PROJECT_DIRECTORY' => $c['project_directory'],
+            'PROJECT_ROOT_DOMAIN' => $c['root_domain'],
+            'PROJECT_DOMAINS' => $domains,
+            'PROJECT_START_WORKERS' => $c['start_workers'],
+            'COMPOSER_CACHE_DIR' => $c['composer_cache_dir'],
+            'PHP_VERSION' => $c['php_version'],
+        ], false)
+    ;
+
+    $command = [
+        'docker',
+        'compose',
+        '-p', $c['project_name'],
+    ];
+
+    foreach ($c['docker_compose_files'] as $file) {
+        $command[] = '-f';
+        $command[] = $c['root_dir'] . '/infrastructure/docker/' . $file;
+    }
+    if ($withBuilder) {
+        $command[] = '-f';
+        $command[] = $c['root_dir'] . '/infrastructure/docker/docker-compose.builder.yml';
```

Comment:
ca pourrait pas etre une variable ce fichier ? genre `$c['docker_compose_builder_files'] = ['docker-compose.builder.yml']`

Reply:
Never seen a project with more than one file for the builder.